store/metrics.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"github.com/coreos/etcd/Godeps/_workspace/src/github.com/prometheus/client_golang/prometheus"
+)
+
+// Set of raw Prometheus metrics.
+// Labels
+// * action = declared in event.go
+// * outcome = Outcome
+// Do not increment directly, use Report* methods.
+var (
+	readCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "etcd",
+			Subsystem: "store",
+			Name:      "reads_total",
+			Help:      "Counter of reads action by (get/getRecursive), outcome (success/failure).",
+		}, []string{"action", "outcome"})
+
+	writeCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "etcd",
+			Subsystem: "store",
+			Name:      "writes_total",
+			Help: "Counter of writes by action (set/delete/update/create/compareAndSwap/compareAndDelete/expire)" +
+				"outcome(success/failure).",
+		}, []string{"action", "outcome"})
+
+	expireCounter = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "etcd",
+			Subsystem: "store",
+			Name:      "expires_total",
+			Help:      "Counter of number of key expirations.",
+		})
+
+	watchRequests = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "etcd",
+			Subsystem: "store",
+			Name:      "watch_requests_total",
+			Help:      "Counter of watch requests incoming into the system.",
+		})
+
+	watcherCount = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "etcd",
+			Subsystem: "store",
+			Name:      "watchers",
+			Help:      "Number of active watchers.",
+		})
+)
+
+type Outcome string
+
+const (
+	Success Outcome = "success"
+	Failure Outcome = "failure"
+)
+
+const (
+	GetRecursive = "getRecursive"
+)
+
+func init() {
+	prometheus.MustRegister(readCounter)
+	prometheus.MustRegister(writeCounter)
+	prometheus.MustRegister(expireCounter)
+	prometheus.MustRegister(watchRequests)
+	prometheus.MustRegister(watcherCount)
+}
+
+func ReportReadRequest(read_action string, outcome Outcome) {
```

Comment:
Renamed them private. 

I used these because for example in the watcher code, it is obvious from the method name that this is just for reporting purposes, and it's not some internal `watcherCount` that is used for realz.

I can remove these, but I actually think it makes the code cleaner, and makes it really easy to find in the codebase where the things are incremented/reported.